bugsnag.go

```diff
@@ -169,10 +171,12 @@ func HandlerFunc(h http.HandlerFunc, rawData ...interface{}) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		request := r
 		// Record a session if auto notify session is enabled
-		if Config.IsAutoCaptureSessions() {
-			ctx := StartSession(r.Context())
-			request = r.WithContext(ctx)
+		ctx := r.Context()
+		if notifier.Config.IsAutoCaptureSessions() {
+			ctx = StartSession(ctx)
 		}
+		ctx = AttachRequestData(ctx, request)
+		request = r.WithContext(ctx)
 		defer notifier.AutoNotify(request)
```

Comment:
Initially I thought it didn't actually do anything. Turns out it did do something, but it was not at all obvious what or where; if the request gets passed in we have a default `httpRequestMiddleware` that would extract request data and set it as metadata, context (`bugsnag.Context`, not `context.Context`), and a default user ID.

I've changed this to now hopefully be more obvious and consistent.

I've left a smaller version of the previous middleware solution because:
1. It adds a map of query parameters to the Request tab in the dashboard (which is not part of the `request` object in the API).
2. It is a good example of how users can write their own custom middleware.